rfc/0003-policy-signing.md

```diff
@@ -0,0 +1,440 @@
+      -----BEGIN PUBLIC KEY-----
+      AUYSDHKJFGSHGSHCAQYIKoZIzj0DAQcDQgAEX0HFTtCfTtPmkx5p1RbtwDE1EVzu
+      wjQs1cCRKb5Pz/yUspkQsN3FO4iyWodCy5j3o0CdIJD/1gvq98pf4IG9tA==
+      -----END PUBLIC KEY-----
+    john:|
+      -----BEGIN PUBLIC KEY-----
+      MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAELKhD7F5OKy77Z582Y6h0u1J3GNA+
+      kvUsh4eKpd1lwkDAzfFDs7yXEExsEkPPuiQJBelDT68n7PDIWB/QEY7mrA==
+      -----END PUBLIC KEY-----
+```
+
+* `keys`: list of keys that are trusted. The owner's name is not relevant during
+  the verification process.
+* `threshold`: an integer number of keys whose signatures are required in order
+  to consider a policy as being properly signed.
+
+Given the above example, there are three keys defined and the threshold is set to
+`2`. That means all the policies loaded must be signed by two or more of these
+keys.
+
+This gives some resiliency if one of these keys is compromised.
+
+### Issuers
+
+It allows to define, on a per issuer basis, who are the trusted`Subject`s:
+
+```yaml
+issuers:
+  "https://token.actions.githubusercontent.com":
+  - https://github.com/flavio/
+  - https://github.com/kubewarden/
+  - https://github.com/alice/fancy-policy/
+  "https://github.com/login/oauth":
+  - john@acme.org
+```
+
+The string entered by the user is then used to match the beginning of the `Subject`
+found inside of a signature.
+
+For example, given the configuration from above:
+
+Subject | Verified | Reason |
+------------------|----------|--------|
+`https://github.com/flavio/policy-secure-pod-images/.github/workflows/release.yml@refs/heads/main` | ✅ | The policy is coming from `https://github.com/flavio/`
+`https://github.com/flavio/psp-capabilities/.github/workflows/release.yml@refs/tags/v0.0.2` | ✅ | The policy is coming from `https://github.com/flavio/`
+`https://github.com/kubewarden/privileged-pods/.github/workflows/release.yml@refs/tags/v0.2.0` | ✅ | The policy is coming from `https://github.com/kubewarden/`
+`https://github.com/kubewarden/psp-capabilities/.github/workflows/release.yml@refs/tags/v0.3.0` | ✅ | The policy is coming from `https://github.com/kubewarden/`
+`https://github.com/alice/fancy-policy/.github/workflows/cd.yml@refs/heads/main` | ✅ | The policy is coming from `https://github.com/alice/fancy-policy/`
+`https://github.com/alice/another-policy/.github/workflows/cd.yml@refs/heads/main` | ❌ | The policy is coming from a repository owned by Alice, but not trusted by the configuration
+
+Given the previous configuration, we would also trust all the policies that have been
+signed with the `john@acme.org` email, as long as the identity provider is the
+GitHub website (this is the identity provider used by "human-driven" keyless signatures
+relying on the GitHub identity provider).
+
+**Note well:** `issuers` defines a per-issuer allow list. That means the policies
+**do not have** to be signed by all the subjects mentioned inside of the list.
+They must all be signed by at least one of these issuers.
```

Comment:
I think this might be confusing. I would prefer to just have tuples of `(issuer, email)`, `issuer, url)`, and use `anyOf`/ `allOf`:

```yaml
anyOf: # at least 1 of all info listed here needs to be verified for accepting
  keyless:
    - issuer: https://token.actions.githubusercontent.com
      url: https://github.com/kubewarden
    - issuer: https://oauth2.suse.com/auth
      email: kubewarden@suse.de
```

Reply:
addressed with the latest rev